pkg/cosign/cosign.go

```diff
@@ -48,37 +49,37 @@ func Initialize(client kubernetes.Interface, namespace, serviceAccount string, i
 	return nil
 }
 
-func VerifySignature(imageRef string, key []byte, repository string, log logr.Logger) (digest string, err error) {
-	pubKey, err := decodePEM(key)
+func VerifySignature(imageRef string, key string, log logr.Logger) (digest string, err error) {
```

Comment:
The repository parameter allows using as separate Cosign repository. See: https://github.com/kyverno/kyverno/issues/2294. How will this be handled?

Reply:
ah see, I think we should set this as an environment variable if users set this repository field within the CR.

:eyes: https://github.com/sigstore/cosign/blob/278ad7d4063592b822ef58bf4a88abee7bb2eff3/cmd/cosign/cli/options/registry.go#L43

Reply:
PTAL @JimBugwadia @Dentrax @erkanzileli, I think the following commit might fix this:
https://github.com/kyverno/kyverno/pull/2607/commits/55a59ba0d4dd99457759f2c5176705c51027157e